### PR TITLE
Doc revisions.

### PR DESCRIPTION
**Pull request by rajkaramchedu**
Clarified the getting started section and added section headers in some places of the guide. The required "match" between SDK, CLI and aptos-core branch can still be confusing but it's not that bad. I will wait until thing settle and maybe draw a picture to explain it. Things are in a flux now so not doing much nitpicking in that area yet. 

Next, I will focus on getting the tutorials with SDK.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2682)

### CODE REVIEW

File: developer-docs-site/docs/cli-tools/aptos-cli-tool/install-aptos-cli.md [hunk 1]
```diff
@@ -10,12 +10,13 @@ This document describes how to install the `aptos` CLI tool. See [Use Aptos CLI]
...
 ## Install precompiled binary (easy mode)
 
-* Navigate to the [release page](https://github.com/aptos-labs/aptos-core/releases) for Aptos CLI.
-* Download the latest release for your platform.
-* Place this at a location for you to run it e.g. `~/bin/aptos` in Linux.
-* On Linux and Mac, make this executable `chmod +x ~/bin/aptos`.
-* Now type `~/bin/aptos help` to read help instructions.
-* If you want you can add `~/bin` to your path in your appropriate `.bashrc` or `.zshrc` for future use.
+1. Navigate to the [release page](https://github.com/aptos-labs/aptos-core/releases?q=cli&expanded=true) for Aptos CLI.
+2. From the latest release section, download the binary zip file for your platform. The binary zip files contain the platform name in the filename, i.e., `aptos-cli-0.2.3-MacOSX-x86_64.zip` for the MacOS binary and `aptos-cli-0.2.3-Ubuntu-x86_64.zip` for the Linux binary.
```
Comment (davidiw) [inline developer-docs-site/docs/cli-tools/aptos-cli-tool/install-aptos-cli.md:14]: considering these versions change so frequently, do we want to be explicit with which version?

File: developer-docs-site/docs/guides/guide-for-system-integrators.md [hunk 2]
```diff
@@ -25,16 +25,29 @@ This guide will overview the following topics for integrating with Aptos:
...
-* Export the appropriate `NODE_URL`: `export NODE_URL=http://127.0.0.1:8080"`.
+To get started, you will need:
+
+1. The Aptos CLI tool.
+2. A local testnet with a validator node that has the faucet service turned on, and
+3. The correct version of the Aptos SDK. The version is important to ensure that the SDK code builds the clients that talk to the local tesnet. 
+
+Follow the below steps to set up the above three:
+
+1. Follow the instructions for Aptos CLI in [Install with cargo](/cli-tools/aptos-cli-tool/install-aptos-cli#install-with-cargo) and install the CLI tool. Make sure you that when you execute Step 2, you use `--branch main` to target the `main` branch, and not the `--branch devnet` as the codebase is undergoing many changes during the next 2-3 weeks.
```
Comment (davidiw) [inline developer-docs-site/docs/guides/guide-for-system-integrators.md:36]: might be good to express why one might use one versus the other:

main branch is for local development -- e.g., local node with faucet
devnet branch is for development on dvent

File: developer-docs-site/docs/guides/guide-for-system-integrators.md [hunk 3]
```diff
@@ -25,16 +25,29 @@ This guide will overview the following topics for integrating with Aptos:
...
+2. Start a [local node with a faucet](/nodes/local-testnet/using-cli-to-run-a-local-testnet#starting-a-local-testnet-with-a-faucet): `aptos node run-local-testnet --with-faucet`. This will See the [Starting a local testnet with a faucet, using CLI](https://aptos.dev/nodes/local-testnet/using-cli-to-run-a-local-testnet#starting-a-local-testnet-with-a-faucet) for more.
+3. Install the SDK from the `aptos-core`. Follow the steps described in [Use the SDK from aptos-core](/guides/local-testnet-dev-flow#typescript-use-the-sdk-from-aptos-core).
```
Comment (davidiw) [inline developer-docs-site/docs/guides/guide-for-system-integrators.md:38]: why only point to the typescript? should point to python too

File: developer-docs-site/docs/guides/guide-for-system-integrators.md [hunk 4]
```diff
@@ -137,7 +150,20 @@ BCS encoded transactions can be submitted to the `/transactions` endpoint but
...
+Within a given transaction, the target of execution can be one of two types: 
+
+- An entry point (formerly known as script function), and/or
+- A script (payload). 
+
+Currently the SDKs: [Python](https://github.com/aptos-labs/aptos-core/blob/76b654b54dcfc152de951a728cc1e3f9559d2729/ecosystem/python/sdk/aptos_sdk/client.py#L248) and [Typescript](https://github.com/aptos-labs/aptos-core/blob/76b654b54dcfc152de951a728cc1e3f9559d2729/ecosystem/typescript/sdk/src/aptos_client.test.ts#L98) only support the generation of transactions that target entry points. This guide points out many of those entry points, such as `coin::transfer` and `account::create_account`. 
+
+All operations on the Aptos blockchain should be available via entry point calls. While one could submit multiple transactions calling entry points in series, many such operations may benefit from being called atomically from a single transaction. 
+
+A script payload transaction can call any entry point or public function defined within any module.
```
Comment (davidiw) [inline developer-docs-site/docs/guides/guide-for-system-integrators.md:162]: feels like they should be together.

**Review by davidiw — COMMENTED** [review]
looking good -- figured we should pass around to align on approach

also one delta between our writing styles is you break things up into distinct sentence / paragraphs. Would like to better understand how that works, because well I am anal :)

File: developer-docs-site/docs/cli-tools/aptos-cli-tool/install-aptos-cli.md (hunk 1 above)
Comment (rajkaramchedu) [reply developer-docs-site/docs/cli-tools/aptos-cli-tool/install-aptos-cli.md:14]: This is just to show them an example of the filename w/ platform name in it, not pointing to a specific version. But the readers might come to the same conclusion as you did, so I will tweak it.

**Comment by rajkaramchedu** [comment]
> looking good -- figured we should pass around to align on approach
> 
> also one delta between our writing styles is you break things up into distinct sentence / paragraphs. Would like to better understand how that works, because well I am anal :)

It's a learned lesson: on the web, if sentences and paragraphs are packed, eyes glaze over and one struggles to keep track of which sentence they are on while reading. Quite different from reading the same packed paragraphs on paper. Hence the rule of thumb I follow is "avoid walls of text", even short climbable walls, haha. I will send you a link to our style guide.

**Comment by rajkaramchedu** [comment]
This doc's evolved enough to have more eyes on it, hence adding Greg and Daniel. 
@banool I will clean up the "Local testnet dev flow" doc separately. However, can you take a look at where I refer to it in the Getting Started section in this doc? 
The goal is to keep merging incremental updates like these and get the system integrator's doc to a final shape, soon. Danke!

**Review by davidiw — APPROVED** [review]

**Pull request merged**